src/time.rs

```diff
@@ -0,0 +1,136 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use core::{ops, fmt};
+
+static MILLIS_PER_SEC: u64 = 1_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TimeStamp {
```

Comment:
Why not use `Instant` and `Duration`, like libstd does?

Reply:
Updated